plenum/server/req_authenticator.py

```diff
@@ -0,0 +1,51 @@
+from typing import Optional
+
+from plenum.common.constants import TXN_TYPE
+from plenum.common.exceptions import NoAuthenticatorFound
+from plenum.common.types import OPERATION
+from plenum.server.client_authn import ClientAuthNr
+
+
+class ReqAuthenticator:
+    """
+    Maintains a list of authenticators. The first authenticator in the list
+    of authenticators is the core authenticator
+    """
+    def __init__(self):
+        self._authenticators = []
+
+    def register_authenticator(self, authenticator: ClientAuthNr):
+        self._authenticators.append(authenticator)
+
+    def authenticate(self, req_data):
```

Comment:
Do we have tests for this method?

Reply:
Yes, in `plenum/test/test_req_authenticator.py`